src/AdventureGoldGovernance.sol

```diff
@@ -0,0 +1,113 @@
+    /// to depositing.
+    /// @dev The AGLD tokens are transferred from the msg.sender to this contract.
+    function deposit(uint256 amount) external {
+        // Follows the Checks-Effects-Interactions pattern
+        // Checks
+        // Allowance and transfer checks occur in the AGLD token contract
+
+        // Effects
+        // Mint the same amount of Adventure Gold Governance tokens
+        _mint(msg.sender, amount);
+
+        // Interactions
+        // Transfer AGLD tokens from the sender to this contract
+        AGLD_TOKEN.safeTransferFrom(msg.sender, address(this), amount);
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice Burns Adventure Gold Governance tokens and withdraws AGLD tokens in return.
+    /// @param amount The amount of Adventure Gold Governance tokens to burn.
+    /// @dev The Adventure Gold Governance tokens are burned from the msg.sender
+    /// and the AGLD tokens are transferred to the msg.sender.
+    function withdraw(uint256 amount) external {
+        // Follows the Checks-Effects-Interactions pattern
+        // Checks
+        // No allowance checks are needed since the msg.sender's tokens are
+        // burned
+
+        // Effects
+        // Burn the same amount of Adventure Gold Governance tokens from the
+        // sender
+        burn(amount);
+
+        // Interactions
+        // Transfer AGLD tokens from this contract to the sender
+        AGLD_TOKEN.safeTransfer(msg.sender, amount);
+
+        emit Withdrawal(msg.sender, amount);
+    }
+
+    /// @notice Only transfers to/from this contract are allowed
+    /// @dev For Solidity lineraization, see https://medium.com/@kalexotsu/inheritance-inheritance-order-and-the-super-keyword-in-solidity-bbe49a2478b6
+    /// @dev ERC20Votes is called since Solidity inheritance is linearized from right to left
+    function _update(address from, address to, uint256 value) internal virtual override(ERC20, ERC20Votes) {
+        if (from != address(this) || to != address(this)) revert OnlyTransfersToFromContract();
+        super._update(from, to, value);
+    }
+
+    /// @notice Overrides IERC6372 functions to make the token & governor timestamp-based
+    function clock() public view override(IERC6372, Votes) returns (uint48) {
+        return uint48(block.timestamp);
+    }
+
+    /// @notice Returns the clock mode as "mode=timestamp".
+    /// @return The clock mode string.
+    // solhint-disable-next-line func-name-mixedcase
+    function CLOCK_MODE() public pure override(IERC6372, Votes) returns (string memory) {
+        return "mode=timestamp";
+    }
+
+    /// @notice Returns the nonce of an owner.
+    /// @param owner The address of the owner.
+    /// @return Nonce value.
+    // The functions below are overrides required by Solidity.
+    function nonces(address owner) public view virtual override(ERC20Permit, Nonces) returns (uint256) {
+        return super.nonces(owner);
+    }
```

Comment:
Does this need to be overriden?

Reply:
As in is this required?

Reply:
It is required -- added the error message that it returns